src/js/components/covid19/recipient/map/MapWrapper.jsx

```diff
@@ -209,7 +210,16 @@ export default class MapWrapper extends React.Component {
 
         // generate the highlight layers that will be shaded in when populated with data filters
         // set up temporary empty filters that will show nothing
-        const colors = this.getColors();
+        let colors = [];
+        if (this.props.data.values.length !== 0) {
+            if (this.props.activeFilters.territory === 'state') {
+                colors = this.getColors(numStateQuantiles);
```

Comment:
lets add colors to state since we are doing this multiple times.

Reply:
I can make two arrays in the state for the class one for the county colors and state colors but those conditional statements will remain to get the appropriate rangeArray based on this.props.activeFilters.territory. Unfortunately, I could not figure out how to render the map without hard coding these values. 

The issue is because in the loadSource() function there is we set up sets of empty filters based on this.props.activeFilters.territory that can later be used to color the map in displayData(). In cases when we had limited reported data because of the selected filters, this was causing issues. 

For example, say we land on the page and get a map with all 50 states colored. Then we select filters such that we only have data for 15 states the behaviour was not as expected because we had previously generated a set of 50 filters and only the first 15 are used. I tried moving out the the code in the loadSource() function that generates the sets of empty filters so in cases like the one I described the map would have a set of 15 colors to use. Im sure this is the most effective method to refactor the code to account for smaller datasets with missing values for some states. However, the when I tried doing it I was creating errors and then there was a lag in rendering the map and then the page would re-render but abnormally, with like a quarter of the map having no color.